test/e2e/networking.go

```diff
@@ -102,160 +103,245 @@ var _ = Describe("Networking", func() {
+//returns true if polling succeeds.
+func pollPeerStatus(f *Framework, svc *api.Service, pollTimeoutSeconds time.Duration) bool {
+	Logf("Begin polling %v %v", svc.Name, pollings)
+	pollings += 1
+	getDetails := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("read").
+			DoRaw()
+	}
+
+	getStatus := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("status").
+			DoRaw()
+	}
 
-		By("Creating a webserver (pending) pod on each node")
+	passed := false
 
-		nodes, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+	Logf("Begin polling " + svc.Name)
+	expectNoError(wait.Poll(2*time.Second, pollTimeoutSeconds, func() (bool, error) {
+		body, err := getStatus()
 		if err != nil {
 			Failf("Failed to list nodes: %v", err)
 		}
-		// previous tests may have cause failures of some nodes. Let's skip
-		// 'Not Ready' nodes, just in case (there is no need to fail the test).
-		filterNodes(nodes, func(node api.Node) bool {
-			return isNodeReadySetAsExpected(&node, true)
-		})
 
-		if len(nodes.Items) == 0 {
-			Failf("No Ready nodes found.")
-		}
-		if len(nodes.Items) == 1 {
-			// in general, the test requires two nodes. But for local development, often a one node cluster
-			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-			// only in some cases
-			if !providerIs("local") {
-				Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", testContext.Provider))
+		// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
+		switch {
+		case string(body) == "pass":
+			passed = true
+			return true, nil
+		case string(body) == "running":
+		case string(body) == "fail":
+			if body, err = getDetails(); err != nil {
+				Failf("Failed on attempt. Error reading details: %v", err)
+				return false, err
+			} else {
+				Failf("Failed on attempt. Details:\n%s", string(body))
+				return false, nil
 			}
-			Logf("Only one ready node is detected. The test has limited scope in such setting. " +
-				"Rerun it with at least two nodes to get complete coverage.")
+		case strings.Contains(string(body), "no endpoints available"):
+			Logf("Attempt: waiting on service/endpoints")
+		default:
+			Logf("Unexpected response:\n%s", body)
 		}
+		return false, nil
+	}))
+
+	if !passed {
+		if body, err := getDetails(); err != nil {
+			Failf("Timed out.  Major error : Couldn't read service details: %v", err)
+		} else {
+			Failf("Timed out. Service details :\n%s", string(body))
+		}
+	}
 
-		podNames := LaunchNetTestPodPerNode(f, nodes, svcname, "1.6")
+	//Just for type safety: But we expect a failure to have occured if the test fails.u
+	return passed
```

Comment:
since `passed` is set to true right before returning true, maybe just eliminate `passed` entirely. the if statement on L193 doesn't need to exist then, either - falling out of the polling loop means there was a failure.